lib/common.go

```diff
@@ -26,18 +26,44 @@ const (
 // the slave knows from which string permutation to start and at which string
 // permutation to stop.
 type Task struct {
-	ID        int
-	JobID     int
-	Algorithm string
-	Salt      string
-	Digest    string
-	CharSet   string
-	Length    int
-	Start     string
-	End       string
+	Job
```

Comment:
Are we able to find the belonging job this way? If eventually a task would finish, how could we find the belonging job? Do we need a `JobID` here?